### PR TITLE
XSUP-41017/AD

### PR DESCRIPTION
**Pull request by sapirshuker**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes:[ link to the issue](https://jira-dc.paloaltonetworks.com/browse/XSUP-41017)

## Description
Fixed an issue where the ***ad-get-user*** command does not pull the accurate PASSWORD_EXPIRED flag due to deprecation of the PASSWORD_EXPIRED flag in the UserAccountControl attribute.
From the [documentation](https://learn.microsoft.com/en-us/troubleshoot/windows-server/active-directory/useraccountcontrol-manipulate-account-properties)

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/Active_Directory_Query/Integrations/Active_Directory_Query/Active_Directory_Query.py [hunk 1]
```diff
@@ -683,6 +695,20 @@ def search_users(default_base_dn, page_size):
...
+                    user_account_to_boolean_fields_msDS_user_account_control_computed(
+                        user_account_control
+                    )
+                )
+                if user.get("userAccountControlFields"):
+                    user["userAccountControlFields"].update(
+                        user_account_to_boolean_dict
+                    )
+                else:
+                    user["userAccountControlFields"] = user_account_to_boolean_dict
```
Comment (jlevypaloalto) [inline Packs/Active_Directory_Query/Integrations/Active_Directory_Query/Active_Directory_Query.py:710]:
```suggestion
                user.setdefault("userAccountControlFields", {}).update(user_account_to_boolean_dict)
```

**Review by jlevypaloalto — COMMENTED** [review]
Looks great!

File: Packs/Active_Directory_Query/Integrations/Active_Directory_Query/Active_Directory_Query.py (hunk 1 above)
Comment (sapirshuker) [reply Packs/Active_Directory_Query/Integrations/Active_Directory_Query/Active_Directory_Query.py:710]: I don't want to set default if the value does not exist.

File: Packs/Active_Directory_Query/Integrations/Active_Directory_Query/Active_Directory_Query_test.py [hunk 2]
```diff
@@ -856,3 +856,43 @@ def modify_dn(self, dn, cn, new_superior):
...
+    mocker_msDSUserAccountControlComputed = mocker.patch.object(
+        Active_Directory_Query,
+        "user_account_to_boolean_fields_msDS_user_account_control_computed",
+        return_value={},
+    )
+
+    Active_Directory_Query.connection = ConnectionMocker()
+
+    Active_Directory_Query.search_users("dc", 1)
+    mocker_msDSUserAccountControlComputed.assert_called_once()
```
Comment (eyalpalo) [inline Packs/Active_Directory_Query/Integrations/Active_Directory_Query/Active_Directory_Query_test.py:898]: Lets check also that the correct value was returned, not only that the method was called.

File: Packs/Active_Directory_Query/ReleaseNotes/1_6_37.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+
+#### Integrations
+
+##### Active Directory Query v2
+
+Fixed an issue where the ***ad-get-user*** command does not pull the accurate PASSWORD_EXPIRED flag.
```
Comment (eyalpalo) [inline Packs/Active_Directory_Query/ReleaseNotes/1_6_37.md:6]:
```suggestion
Fixed an issue where the ***ad-get-user*** command failed to retrieve the correct PASSWORD_EXPIRED flag.
```

**Review by eyalpalo — CHANGES_REQUESTED** [review]

File: Packs/Active_Directory_Query/Integrations/Active_Directory_Query/Active_Directory_Query.py [hunk 4]
```diff
@@ -193,6 +193,17 @@ def user_account_to_boolean_fields(user_account_control):
...
 
+def user_account_to_boolean_fields_msDS_user_account_control_computed(user_account_control):
+    """
+    parse the msDS-User-Account-Control-Computed into boolean values.
+    following the values from:
+    https://learn.microsoft.com/en-us/windows/win32/adschema/a-msds-user-account-control-computed
+    """
+    return {
+        'PASSWORD_EXPIRED': bool(user_account_control & 0x800000),
+    }
```
[the remark this answers is not in the document]
Comment (sapirshuker) [reply Packs/Active_Directory_Query/Integrations/Active_Directory_Query/Active_Directory_Query.py:204]: There are more values for this attribute but they are new values and were not in the original integration.

[the remark this answers is not in the document]
Comment (sapirshuker) [reply Packs/Active_Directory_Query/Integrations/Active_Directory_Query/Active_Directory_Query.py:205]: Added

File: Packs/Active_Directory_Query/Integrations/Active_Directory_Query/Active_Directory_Query.py (hunk 1 above)
Comment (jlevypaloalto) [reply Packs/Active_Directory_Query/Integrations/Active_Directory_Query/Active_Directory_Query.py:713]: It's just a shorter way of writing those lines, the effect is the same.

**Comment by sapirshuker** [comment]
XSIAM step failed on L#636 on the build status this is on-call known issue -
TPBs Failure on XSIAM machines - incidents are not created

**Review by eyalpalo — APPROVED** [review]

**Comment by sapirshuker** [comment]
XSIAM step failed on L#636 on the build status this is on-call known issue -
TPBs Failure on XSIAM machines - incidents are not created
@ShahafBenYakir please force merge.

**Pull request merged**